### PR TITLE
feat(whiskers): write results to `filename` in single file mode

### PR DESCRIPTION
**Pull request by uncenter**
Now writes the results of single file templates to `whiskers.filename` if present, like in matrix mode.

### CODE REVIEW

**Review by sgoudham — CHANGES_REQUESTED** [review]
I think it'd be good to update the `justfiles` in the [single-file](https://github.com/catppuccin/toolbox/tree/main/whiskers/examples/single-file) directory as part of this PR.

File: whiskers/src/main.rs [hunk 1]
```diff
@@ -327,21 +357,8 @@ fn render_multi_output(
...
 
-        if args.dry_run || cfg!(test) {
-            println!(
-                "Would write {} bytes into {}",
-                result.as_bytes().len(),
-                filename.display()
-            );
-        } else if args.check.is_some() {
-            check_result_with_file(&filename, &result).context("Check mode failed")?;
-        } else {
```
Comment (sgoudham) [inline whiskers/src/main.rs:340]: I believe there's a regression here:

This if statement is not present in the extracted `write_template` function defined above.

**Review by sgoudham — CHANGES_REQUESTED** [review]

File: whiskers/src/main.rs [hunk 2]
```diff
@@ -294,13 +325,14 @@ fn render_single_output(
...
     ctx: &tera::Context,
-    palette: &models::Palette,
     tera: &tera::Tera,
+    matrix: HashMap<String, Vec<String>>,
+    palette: &models::Palette,
+    filename_template: &str,
     template_name: &str,
-    args: &Args,
+    dry_run: bool,
+    check: bool,
```
Comment (nullishamy) [inline whiskers/src/main.rs:335]: Any particular reason to change these? Seems like it would be more useful to have them wrapped in the struct?

**Review by nullishamy — APPROVED** [review]
LGTM, seems to have conflicts? Can get those fixed and then this is good for merge :+1:

**Comment by uncenter** [comment]
Ah yeah it was the trailing newlines I fixed in my last PR. Should be good now.

**Review by nullishamy — APPROVED** [review]

**Review by backwardspy — APPROVED** [review]

**Review by sgoudham — APPROVED** [review]

**Pull request merged**